src/SlateEditor.component.js

```diff
@@ -76,14 +76,17 @@ function setLinkByKey(editor, nodeKey, href, openInNewWindow) {
 }
 
 /**
- * 修改 image node 的 alt
+ * 修改 image node 的 alt 和連結
  * @param {object} editor
  * @param {*} nodeKey slate editor 針對每一個 node 產生的 unique key
  * @param {string} alt image 的 alt attribute
  * @param {string} src src image 的 src
+ * @param {string} href 超連結
+ * @param {boolean} openInNewWindow 是否要開新視窗
  */
-function setImageAltByKey(editor, nodeKey, alt, src) {
-  const data = { alt, src };
+function setImageInfoByKey(editor, nodeKey, alt, src, href, openInNewWindow) {
+  const data = { alt, src, href };
+  if (openInNewWindow) { data.target = '_blank'; }
```

Comment:
可以再補個 `rel='noopener noreferrer'`
https://github.com/hahow/hh-frontend-react/pull/3947

```suggestion
  if (openInNewWindow) {
    data.target = '_blank';
    data.rel = 'noopener noreferrer';
  }
```

Reply:
這部分我偏好在 [serialize](https://github.com/hahow/hh-slate-editor/pull/78/files#diff-8b89036141479d8f8861d8e4bd4f32d61973f2f610a7e37a8312d78b23f81c25R132) 的過程中處理，`setImageInfoByKey` 動作越單純越好，因為這邊只是把值往 `node.data` 裡塞。目前是這個照著這個專案內處理連結的方式實作，我猜前人這麼寫只是為了讓 data 內的值能直接對應上 tag attribute。

我目前的想法是這個版本可以先照上方的方式處理（只更動圖片連結的部分），然後再一個 refactor 分支翻其他連結。但實際怎麼處理這段比較好，下面開放討論串XD

Reply:
結果我最後還是直接在 serializer 裡頭處理掉了 cec18f8